src/main.cpp

```diff
@@ -3729,6 +3737,13 @@ bool ActivateBestChain(CValidationState& state, const CChainParams& chainparams,
 {
     CBlockIndex *pindexMostWork = NULL;
     CBlockIndex *pindexNewTip = NULL;
+
+    // ABC maintains a fair degree of expensive-to-calculate internal state
+    // because this function periodically releases cs_main so that it does not lock up other threads for too long
+    // during large connects - and to allow for e.g. the callback queue to drain
+    // we use m_cs_chainstate to enforce mutual exclusion so that only one caller may execute this function at a time
+    LOCK(m_cs_chainstate);
+
```

Comment:
Note to self: https://github.com/bitcoin/bitcoin/pull/13023/commits/a3ae8e68739023e5dba9e5cb190e707ed4603316 has this diff above the definitions of `pindexMostWork` and `pindexNewTip` instead of below.

Non-blocking; fix this during the rebase if you want.